### PR TITLE
[opt](inverted index) mow supports index optimization

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Proposed changes

1. mow support no need_read_data opt.
2. mow support topn opt.

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -2233,7 +2233,11 @@ Status SegmentIterator::_read_columns_by_rowids(std::vector<ColumnId>& read_colu
     }
 
     for (auto cid : read_column_ids) {
-        if (_prune_column(cid, (*mutable_columns)[cid], true, select_size)) {
+        auto& colunm = (*mutable_columns)[cid];
+        if (_no_need_read_key_data(cid, colunm, select_size)) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:2237]: can you call _no_need_read_key_data inside _prune_column?

**Review by xiaokang — COMMENTED** [review]
pls add testcase

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -2892,11 +2919,24 @@ bool SegmentIterator::_can_opt_topn_reads() const {
         return false;
     }
 
-    if (!_col_predicates.empty() || !_col_preds_except_leafnode_of_andnode.empty()) {
+    if (!_common_expr_ctxs_push_down.empty() || !_remaining_conjunct_roots.empty()) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:2922]: Is the modifications in _can_opt_topn_reads related to mow need_read_key_data?

**Review by airborne12 — DISMISSED** [review]
LGTM

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**